src/device/pf_fspm.c

```diff
@@ -197,6 +229,81 @@ int pf_fspm_validate_configuration (const pnet_cfg_t * p_cfg)
    return 0;
 }
 
+/**
+ * @internal
+ * Load the logbook from nonvolatile memory, if existing.
+ *
+ * @param net              InOut: The p-net stack instance
+ */
+static void pf_fspm_load_logbook (pnet_t * net)
+{
+   const char * p_file_directory = NULL;
+
+   (void)pf_cmina_get_file_directory (net, &p_file_directory);
+
+   if (
+      pf_file_load (
+         p_file_directory,
+         PNET_FILENAME_LOGBOOK,
+         &net->fspm_log_book,
+         sizeof (pf_log_book_t)) == 0)
+   {
+      LOG_DEBUG (PNET_LOG, "FSPM(%d): Did read logbook from nvm.\n", __LINE__);
+   }
+   else
+   {
+      LOG_DEBUG (
+         PNET_LOG,
+         "FSPM(%d): Could not yet read logbook from nvm.\n",
+         __LINE__);
+      memset (&net->fspm_log_book, 0, sizeof (pf_log_book_t));
+   }
+}
+
+/**
+ * @internal
+ * Save the logbook to nonvolatile memory, if necessary.
+ *
+ * Compares with the content of already stored logbook (in order not to
+ * wear out the flash chip)
+ *
+ * @param net              InOut: The p-net stack instance
+ */
+static void pf_fspm_save_logbook_if_modified (pnet_t * net)
```

Comment:
I would name the function  "pf_fspm_save_logbook" and skip details on flash wear out on this level. This is handled by the underlying write operation but does not need to be visible in this api.

This is comment only.

Reply:
Done.

Reply:
ok